src/main/java/com/sapred/ordermanagerred/service/UserService.java

```diff
@@ -30,79 +32,61 @@ public class UserService {
         if(!authenticatedUserEmail.getPassword().equals(password))
-            return new ResponseEntity<>("Unauthorized", HttpStatus.UNAUTHORIZED); // 401
+            throw new ResponseStatusException(HttpStatus.UNAUTHORIZED);
         String token = jwtToken.generateToken(authenticatedUserEmail);
-        return new ResponseEntity<>(token, HttpStatus.OK);
+        return token;
     }
 
 
-    public ResponseEntity<Boolean> deleteCustomer(String token, String customerId) {
+    @SneakyThrows
+    public void deleteUser(String token, String userId) {
         RoleOptions role = jwtToken.getRoleIdFromToken(token);
         String companyIdFromToken = jwtToken.getCompanyIdFromToken(token);
-        User user = userRepository.findById(customerId).get();
-        // אם הוא לא מנהל או שהוא לא מהחברה של הלקוח שאותו הוא רוצה למחוק
+        User user = userRepository.getById(userId);
         if (role == RoleOptions.CUSTOMER || !user.getCompanyId().getId().equals(companyIdFromToken))
-            return new ResponseEntity<>(false, HttpStatus.NOT_FOUND);
-        try {
-            userRepository.deleteById(customerId);
-            return new ResponseEntity<>(true, HttpStatus.OK);
-        } catch (RuntimeException e) {
-            return new ResponseEntity<>(false, HttpStatus.INTERNAL_SERVER_ERROR);
-        } catch (Exception e) {
-            return new ResponseEntity<>(false, HttpStatus.NOT_FOUND);
-        }
+            throw new NoPermissionException("You do not have the appropriate permission to delete user");
+        userRepository.deleteById(userId);
     }
 
-    public ResponseEntity<Boolean> editCustomer(String token, UserDTO customer) {
+    @SneakyThrows
+    public void editUser(String token, User user) {
         RoleOptions role = jwtToken.getRoleIdFromToken(token);
         String companyIdFromToken = jwtToken.getCompanyIdFromToken(token);
-        User user = userRepository.findById(customer.getId()).get();
-        // אם הוא לא מנהל או שהוא לא מהחברה של הלקוח שאותו הוא רוצה לערוך
+        User userTOEdit = userRepository.getById(user.getId());
         if (role == RoleOptions.CUSTOMER || !user.getCompanyId().getId().equals(companyIdFromToken))
-            return new ResponseEntity<>(false, HttpStatus.NOT_FOUND);
-        try {
-            user.setFullName(customer.getFullName());
-            Address address = new Address(customer.getPhone(), customer.getAddress(), customer.getEmail());
-            user.setAddress(address);
-            user.getAuditData().setUpdateDate(new Date());
-            userRepository.save(user);
-            return new ResponseEntity<>(true, HttpStatus.OK);
-        } catch (RuntimeException e) {
-            return new ResponseEntity<>(false, HttpStatus.INTERNAL_SERVER_ERROR);
-        } catch (Exception e) {
-            return new ResponseEntity<>(false, HttpStatus.NOT_FOUND);
-        }
+            throw new NoPermissionException("You do not have the appropriate permission to edit user");
```

Comment:
there are more correct ways to do the update 
please read here
https://medium.com/geekculture/types-of-update-operations-in-mongodb-using-spring-boot-11d5d4ce88cf